### PR TITLE
Bug fix in application

### PR DESCRIPTION
**Pull request by Mumsfilibaba**
Fixes a dependency between CommonApplication and PlatformApplication that caused a crash when calling CommonApplication::Terminate()

### CODE REVIEW

File: LambdaEngine/Source/Application/Win32/Win32Application.cpp [hunk 1]
```diff
@@ -193,6 +198,11 @@ namespace LambdaEngine
 		constexpr uint32 REPEAT_KEY_MASK	= 0x40000000;
 		constexpr uint16 BACK_BUTTON_MASK	= 0x0001;
 
+		//if (!m_EventHandler)
+		//{
+		//	return;
+		//}
+
```
Comment (TheoBerlin) [inline LambdaEngine/Source/Application/Win32/Win32Application.cpp]:
```suggestion
```

File: LambdaEngine/Source/Application/API/CommonApplication.cpp [hunk 2]
```diff
@@ -317,8 +309,12 @@ namespace LambdaEngine
 
 	bool CommonApplication::PostRelease()
 	{
-		s_CommonApplication->ReleasePlatform();
+		Application* pPlatformApplcation = s_CommonApplication->GetPlatformApplication();
+		pPlatformApplcation->SetEventHandler(nullptr);
+		
 		s_CommonApplication.Reset();
+
+		SAFEDELETE(pPlatformApplcation);
```
Comment (skumbananaaa) [inline LambdaEngine/Source/Application/API/CommonApplication.cpp]:
```suggestion
		SAFEDELETE(pPlatformApplication);
```

File: LambdaEngine/Source/Application/API/CommonApplication.cpp [hunk 3]
```diff
@@ -280,9 +265,16 @@ namespace LambdaEngine
...
+		Application* pPlatformApplcation = PlatformApplication::CreateApplication();
+		if (!pPlatformApplcation->Create())
+		{
+			return false;
+		}
+
 		// Create application
 		CommonApplication* pApplication = DBG_NEW CommonApplication();
-		if (!pApplication->Create())
+		if (!pApplication->Create(pPlatformApplcation))
```
Comment (skumbananaaa) [inline LambdaEngine/Source/Application/API/CommonApplication.cpp]:
```suggestion
		if (!pApplication->Create(pPlatformApplication))
```

File: LambdaEngine/Source/Application/API/CommonApplication.cpp [hunk 4]
```diff
@@ -317,8 +309,12 @@ namespace LambdaEngine
 
 	bool CommonApplication::PostRelease()
 	{
-		s_CommonApplication->ReleasePlatform();
+		Application* pPlatformApplcation = s_CommonApplication->GetPlatformApplication();
```
Comment (skumbananaaa) [inline LambdaEngine/Source/Application/API/CommonApplication.cpp]:
```suggestion
		Application* pPlatformApplication = s_CommonApplication->GetPlatformApplication();
```

File: LambdaEngine/Source/Application/API/CommonApplication.cpp [hunk 5]
```diff
@@ -280,9 +265,16 @@ namespace LambdaEngine
 
 	bool CommonApplication::PreInit()
 	{
+		// Create platform application
+		Application* pPlatformApplcation = PlatformApplication::CreateApplication();
+		if (!pPlatformApplcation->Create())
```
Comment (skumbananaaa) [inline LambdaEngine/Source/Application/API/CommonApplication.cpp]:
```suggestion
		Application* pPlatformApplication = PlatformApplication::CreateApplication();
		if (!pPlatformApplication->Create())
```

File: LambdaEngine/Source/Application/API/CommonApplication.cpp [hunk 6]
```diff
@@ -317,8 +309,12 @@ namespace LambdaEngine
 
 	bool CommonApplication::PostRelease()
 	{
-		s_CommonApplication->ReleasePlatform();
+		Application* pPlatformApplcation = s_CommonApplication->GetPlatformApplication();
+		pPlatformApplcation->SetEventHandler(nullptr);
```
Comment (skumbananaaa) [inline LambdaEngine/Source/Application/API/CommonApplication.cpp]:
```suggestion
		pPlatformApplication->SetEventHandler(nullptr);
```

**Review by skumbananaaa — CHANGES_REQUESTED** [review]
Minor Spelling issue

**Review by skumbananaaa — APPROVED** [review]

**Pull request merged**